test_crates/inherent_method_now_doc_hidden/new/src/doc_hidden_struct_inherent_method_now_doc_hidden.rs

```diff
@@ -0,0 +1,14 @@
+pub struct Foo;
+
+#[doc(hidden)]
```

Comment:
So, this false positive test fails! It's the one thing that fails.

I tried adding it to the `inherent_method_const_removed` test crate, lint and it also has no effect on output. Is the item `visibility_limit` restriction not working? Or did I do this wrong?

Reply:
The failure I get for this lint:

```
Differences (-expected|+actual):
             "method_visibility": String("public"),
             "name": String("Foo"),
             "path": List([
                 String("inherent_method_now_doc_hidden"),
+                String("doc_hidden_struct_inherent_method_now_doc_hidden"),
+                String("Foo"),
+            ]),
+            "span_begin_line": Uint64(5),
+            "span_filename": String("src/doc_hidden_struct_inherent_method_now_doc_hidden.rs"),
+            "visibility_limit": String("public"),
+        },
+        {
+            "method_name": String("method"),
+            "method_visibility": String("public"),
+            "name": String("Foo"),
+            "path": List([
+                String("inherent_method_now_doc_hidden"),
+                String("doc_hidden_struct_inherent_method_now_doc_hidden"),
+                String("Foo"),
+            ]),
+            "span_begin_line": Uint64(9),
+            "span_filename": String("src/doc_hidden_struct_inherent_method_now_doc_hidden.rs"),
+            "visibility_limit": String("public"),
+        },
+        {
+            "method_name": String("associated_fn"),
+            "method_visibility": String("public"),
+            "name": String("Foo"),
+            "path": List([
+                String("inherent_method_now_doc_hidden"),
                 String("enum_inherent_method_now_doc_hidden"),
                 String("Foo"),
             ]),
             "span_begin_line": Uint64(6),
```

Reply:
Maybe the struct was supposed to be doc hidden rather than the impl, but it still fails if I do that. Or if I do both, fwiw.

Reply:
@DeluxeRaph and I were just chatting about this -- we aren't sure what the semantics of "there's `#[doc(hidden)]` on the _impl itself_" are like. To figure out what _should_ happen here, we need to see what HTML rustdoc generates for cases where the `impl` is hidden, and compare that to when its items are hidden.

I don't think any of our lints currently have tests for `#[doc(hidden)]` on the impl. We only test for `#[doc(hidden)]` on the struct/enum/union, or on the contents of the impl individually.

We don't currently have a lint for `#[doc(hidden)]` on the impl (and it isn't clear how we'd write one). I think it's fine to leave that as future work.

For the sake of this test, I would just move the `#[doc(hidden)]` onto the struct itself and remove it from the `impl` block. Based on your query, I think that will work correctly.

`visibility_limit` just refers to the struct's declaration directly: `pub struct Foo;` means the `visibility_limit` is `"public"`. It doesn't consider `#[doc(hidden)]` or anything else, and appears to be working correctly.

Reply:
> Based on your query, I think that will work correctly.

Funny, I thought I had tried that and it failed (as I mentioned above). But I tried again and you're right. I must have forgotten to rebuild the docs the first time.